app/src/main/java/com/kabouzeid/gramophone/model/lyrics/Lyrics.java

```diff
@@ -0,0 +1,68 @@
+package com.kabouzeid.gramophone.model.lyrics;
+
+import com.kabouzeid.gramophone.model.Song;
+
+import java.util.ArrayList;
+
+public class Lyrics {
+    private static final ArrayList<Class<? extends Lyrics>> FORMATS = new ArrayList<>();
+
+    public Song song;
+    public String data;
+
+    boolean parsed = false;
+    boolean valid = false;
+
+    public Lyrics setData(Song song, String data) {
+        this.song = song;
+        this.data = data;
+        return this;
+    }
+
+    public static Lyrics parse(Song song, String data) {
+        for (Class<? extends Lyrics> format : Lyrics.FORMATS) {
+            try {
+                Lyrics lyrics = format.newInstance().setData(song, data);
```

Comment:
I was more so concerned with something like a person repeatedly skipping songs, so a lot of objects would be instantiated. I guess it's not a huge impact though, like you said.